test/cloudtest/test_k8s_node_recovery.py

```diff
@@ -0,0 +1,303 @@
+
+    suspended_node_name = suspend_first_node_of_pod(mz, cluster_setup.storage_name)
+
+    # expect (stale) data to be delivered timely
+    validate_state(
+        mz,
+        cluster_setup,
+        reached_index,
+        must_exceed_reached_index=False,
+        timeout_in_sec=30,
+    )
+
+    print("Expecting stalled data...")
+    # only request this index because the previous validation succeeded / did not block
+    stalled_index = get_current_counter_index(mz, cluster_setup)
+
+    print("Awaiting node recovery...")
+    # expect live data to be delivered at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        cluster_setup,
+        stalled_index,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
+    )
+
+    recovered_index = get_current_counter_index(mz, cluster_setup)
+
+    mz.revive_suspended_k8s_node(suspended_node_name)
+
+    # expect no issues after node recovery
+    validate_state(
+        mz,
+        cluster_setup,
+        recovered_index + REQUIRED_PROGRESS,
+        must_exceed_reached_index=True,
+        timeout_in_sec=10,
+    )
+
+
+@pytest.mark.long
+@unittest.skip("Requires #20299 to work")
+def test_unreplicated_compute_cluster_on_failing_node(
+    mz: MaterializeApplication,
+) -> None:
+    """
+    An unreplicated compute cluster is on the failed node. Queries of indexes on the compute cluster should fail, but
+    resolve within a minute or two.
+    """
+    cluster_setup = ClusterSetup()
+    populate(mz, cluster_setup)
+    reached_index = get_current_counter_index(mz, cluster_setup)
+
+    suspended_node_name = suspend_first_node_of_pod(mz, cluster_setup.compute_name)
+
+    print("Awaiting node recovery...")
+    # expect (live) data to be delivered after at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        cluster_setup,
+        reached_index + REQUIRED_PROGRESS,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
+    )
+
+    recovered_index = get_current_counter_index(mz, cluster_setup)
+
+    mz.revive_suspended_k8s_node(suspended_node_name)
+
+    # expect no issues after node recovery
+    validate_state(
+        mz,
+        cluster_setup,
+        recovered_index + REQUIRED_PROGRESS,
+        must_exceed_reached_index=True,
+        timeout_in_sec=10,
+    )
+
+
+@pytest.mark.long
+def test_replicated_compute_cluster_on_failing_node(mz: MaterializeApplication) -> None:
+    """
+    A replicated compute cluster is on the failed node. Queries of indexes on the compute cluster should experience no
+    disruption in latency, thanks to the second replica.
+    """
+    cluster_setup = ClusterSetup(compute_replication_factor=2)
+    populate(mz, cluster_setup)
+    reached_index = get_current_counter_index(mz, cluster_setup)
+
+    suspended_node_name = suspend_first_node_of_pod(mz, cluster_setup.compute_name)
```

Comment:
I tried to mitigate that by using different availability zones for storage and compute. But yeah, if I have to specify two zones per cluster as I understand from your previous comment, this will not work with three AZs in total.

Reply:
No, I meant something like this:

1. AZ 1 has 2 K8s nodes. We put one compute replica there
2.  AZ 2 has 1 K8s node. We put the second compute replica and the storage cluster there.
3. We disrupt the compute replica in AZ 1 only. The rest of the stuff is all safe in AZ 2 and is not impacted.